### ISSUE TITLE
Performance issue

### ISSUE THREAD

**Issue by maulik9898** [tonyofrancis/Fetch#395]
I am using this library in one of my project. But there is a serious performance issue with this library. 
To check is this library issue or another issue I compare the performance of this library with another library [Okdownloader](https://github.com/lingochamp/okdownload) and Android app [ADM downloader](https://play.google.com/store/apps/details?id=com.dv.adm&hl=en_IN).

Here is a common configuration:

**Max thread**:9
**Max concurrent Download**:3
**Active Download**:3
**Download URL**:https://excellmedia.dl.sourceforge.net/project/xiaomi-eu-multilang-miui-roms/xiaomi.eu/MIUI-STABLE-RELEASES/MIUIv10/xiaomi.eu_multi_HMNote7Pro_V10.3.5.0.PFHCNXM_v10-9.zip

**Device Info**:https://valid.x86.fr/a/5lfebk

**1) ADM Downloader** ***( CPU Usage: 25% )***

![image](https://user-images.githubusercontent.com/21076623/59836823-db345580-9369-11e9-90e2-cff037fce736.png)

**2) OkDownloader Library** ***( CPU Usage: 24% )***

![image](https://user-images.githubusercontent.com/21076623/59836954-0e76e480-936a-11e9-8c02-9c15bba0cb99.png)

**3) Fetch Library** ***( CPU Usage: 50% )***

![image](https://user-images.githubusercontent.com/21076623/59837005-22224b00-936a-11e9-94ef-2fd0b305757b.png)

I love this library and that's the reason I don't want to move to another library.
But this performance issue is a real pain. I hope u will look into this.

**Comment by maulik9898** [comment]
The App I develop to perform this test https://github.com/maulik9898/PerformanceTest

**Comment by maulik9898** [comment]
Here is CPU profile image from android studio.
1) Ok downloader 

![image](https://user-images.githubusercontent.com/21076623/59837865-99a4aa00-936b-11e9-8984-3545e79ffc53.png)


2) Fetch

![image](https://user-images.githubusercontent.com/21076623/59837900-ae813d80-936b-11e9-92e1-bc54fc44eff5.png)


I am no expert in this but this may be helpful to you.

**Comment by tonyofrancis** [comment]
looking into this

**Comment by maulik9898** [comment]
This problem only present in PARALLEL mode

**Comment by tonyofrancis** [comment]
@maulik9898 Thanks for reporting in detail. Made it super simple to pin point the issue. It is fixed now. It will be in the next release.

**Comment by tonyofrancis** [comment]
@maulik9898 a new version of fetch is available with this fix. see the readme and changelog.

**Comment by maulik9898** [comment]
@tonyofrancis that's awesome man.

**Comment by maulik9898** [comment]
cpu usage of fetchActivity drop from 50% to 20%-25% that's amazing

**Comment by DHosseiny** [comment]
> @maulik9898 a new version of fetch is available with this fix. see the readme and changelog.

Hi! can u tell the version number?

**Comment by tonyofrancis** [comment]
@DHosseiny fixed in androidx version 3.1.2 latest. The bug has existed for a long time now. Glad @maulik9898 spotted it.
